### PR TITLE
Fixes for osx

### PR DESCRIPTION
**Pull request by Tobias-Fischer**

### CODE REVIEW

File: boa/build.py [hunk 1]
```diff
@@ -100,6 +100,8 @@
     shell_path = "/bin/sh"
 elif utils.on_win:
     shell_path = "bash"
+elif "darwin" in sys.platform:
+    shell_path = "/bin/zsh"
```
Comment (wolfv) [inline boa/build.py:104]: how about using `/bin/bash` here? it seems to be still available on OS X?

File: boa/build.py [hunk 2]
```diff
@@ -468,6 +470,7 @@ def write_build_scripts(m, script, build_file):
 
     with open(work_file, 'w') as bf:
         # bf.write('set -ex\n')
+        bf.write('#!/usr/bin/env bash\n\n')
```
Comment (wolfv) [inline boa/build.py:473]: then we could leave out this line

File: boa/build.py [hunk 3]
```diff
@@ -100,6 +100,8 @@
     shell_path = "/bin/sh"
 elif utils.on_win:
     shell_path = "bash"
+elif "darwin" in sys.platform:
```
Comment (wolfv) [inline boa/build.py:103]: also I think there is `utils.on_osx`

File: boa/build.py (hunk 1 above)
Comment (Tobias-Fischer) [reply boa/build.py:104]: This is definitely needed, otherwise compctl is not found.

File: boa/build.py (hunk 2 above)
Comment (Tobias-Fischer) [reply boa/build.py:473]: This line might not be needed actually, I'll try next week

**Comment by wolfv** [comment]
hey sorry, I think the real issue was in the catkin patch.

our previous check was testing what the __default__ shell is. This one checks what the current shell is:

```
if [[ ! -z "$ZSH_VERSION" ]]; then
  CATKIN_SHELL=zsh
elif [[ ! -z "$BASH_VERSION" ]]; then
  CATKIN_SHELL=bash
else
  CATKIN_SHELL=sh
fi
```

File: boa/build.py (hunk 1 above)
Comment (wolfv) [reply boa/build.py:104]: yeah that was an issue with our way of detecting shells in catkin's setup.sh :) luckily!

**Comment by Tobias-Fischer** [comment]
I see. So can this PR be closed then?

**Comment by wolfv** [comment]
yeah, I think unfortunately yes :)

**Pull request closed without merging**